libs/ddhub-client-gateway-frontend/ui/channels/src/lib/containers/modals/Create/Restrictions/effects/roleRestriction.effects.ts

```diff
@@ -37,17 +38,34 @@ export const useRolesRestrictionEffects = (currentRoles: string[]) => {
   };
 
   const addRole = (role: string) => {
-    if (roles.includes(role)) {
-      return;
+    if (!roles.includes(role)) {
+      const rolesToSet = [role, ...roles];
+      setRoles(rolesToSet);
     }
-    setRoles([...roles, role]);
+
     clearRolesInput();
   };
 
   const removeRole = (value: string) => {
     setRoles(roles.filter((role) => role !== value));
   };
 
+  const updateRole = (type: RestrictionType, oldValue: string, newValue: string) => {
```

Comment:
this method looks similar to what you have in didRestriction. Probably you could create new effect with stored state like:
```
useRestrictionEffects() => { 
  const [restriction, updateRestriction] = useState<string[]>([]);
 ...
}
```
and then reuse it in useDID and useRoles restrictions effects. How does it sounds?

Reply:
I agree if create a new effect and contains shared methods. Not too sure about stored state. We'll keep the update function in both useDID and useRoles and then call updateRestriction?

Reply:
if you think that making stored state there make some problems and mess up code, then don't do it. You can just move it to effect/regular function.